packages/manager/apps/pci-kubernetes/src/pages/admission-plugin/AdmissionPlugins.page.tsx

```diff
@@ -0,0 +1,220 @@
+import {
+  OsdsSwitch,
+  OsdsSwitchItem,
+  OsdsButton,
+  OsdsModal,
+  OsdsTooltip,
+  OsdsDivider,
+  OsdsText,
+  OsdsMessage,
+  OsdsTooltipContent,
+  OsdsSpinner,
+} from '@ovhcloud/ods-components/react';
+import { useCallback, useState } from 'react';
+import { useNavigate, useParams } from 'react-router-dom';
+import { ODS_THEME_COLOR_INTENT } from '@ovhcloud/ods-common-theming';
+
+import {
+  ODS_BUTTON_VARIANT,
+  ODS_MESSAGE_TYPE,
+  ODS_SPINNER_SIZE,
+  ODS_SWITCH_SIZE,
+  ODS_TEXT_COLOR_INTENT,
+  ODS_TEXT_LEVEL,
+  ODS_TEXT_SIZE,
+  OdsSwitchChangedEventDetail,
+  OsdsSwitchCustomEvent,
+} from '@ovhcloud/ods-components';
+import { useTranslation } from 'react-i18next';
+import { useNotifications } from '@ovh-ux/manager-react-components';
+import { useKubernetesCluster } from '@/api/hooks/useKubernetes';
+import { useUpdateAdmissionPlugin } from '@/api/hooks/useAdmissionPlugin/useAdmissionPlugin';
+import { useResponsiveModal } from '@/hooks/useResizeOsdsModal';
+
+const AdmissionPluginsModal = () => {
+  const navigate = useNavigate();
+  const onClose = () => navigate('..');
+  const { projectId, kubeId } = useParams();
+  const { data: kubeDetail, isPending } = useKubernetesCluster(
+    projectId,
+    kubeId,
+  );
+  const { t } = useTranslation(['service']);
+
+  const { plugins } = kubeDetail ?? {};
+
+  const [pluginData, setPluginData] = useState(plugins);
+
+  useResponsiveModal('450px');
+  const { addError, addSuccess } = useNotifications();
+
+  const handleChange = useCallback(
+    (e: OsdsSwitchCustomEvent<OdsSwitchChangedEventDetail>, name: string) => {
+      const state = e.detail.current;
+
+      setPluginData((prevPluginData) =>
+        prevPluginData.map((plugin) => {
+          if (plugin.name === name) {
+            return { ...plugin, state };
+          }
+          return plugin;
+        }),
+      );
+    },
+    [],
+  );
+
+  const {
+    updateAdmissionPlugins,
+    isPending: isMutationUpdating,
+  } = useUpdateAdmissionPlugin({
+    projectId,
+    kubeId,
+    onError: (error) => {
+      addError(
+        t('kube_service_cluster_admission_plugins_error', {
+          message: error?.message || null,
+        }),
+        true,
+      );
+      onClose();
+    },
+    onSuccess: () => {
+      addSuccess(t('kube_service_cluster_admission_plugins_success'), true);
+      onClose();
+    },
+  });
+
+  const handleCancel = () => {
+    setPluginData(plugins);
+    onClose();
+  };
+
+  const handleSave = () => {
+    const initialState = { enabled: [], disabled: [] };
+
+    const reorderPlugin = pluginData.reduce((acc, item) => {
+      acc[item.state].push(item.name);
+      return acc;
+    }, initialState);
```

Comment:
We can prevent from mutability:

```suggestion
    const reorderPlugin = pluginData.reduce(
      (acc, item) => ({ ...acc, [item.state]: item.name }),
      initialState,
    );
```